File: EIDSS v5/android.java/workspace/EIDSS/src/com/bv/eidss/EidssAndroidHelpers.java
package com.bv.eidss;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.AlertDialog.Builder;
import android.app.Dialog;
import android.content.DialogInterface;

public class EidssAndroidHelpers {
    public static Dialog ErrorDialog(Activity context, String text)
    {
    	Builder builder = new AlertDialog.Builder(context);
        builder.setMessage(text);
        builder.setCancelable(false);
        builder.setPositiveButton(context.getResources().getString(R.string.Ok), new DialogInterface.OnClickListener() {
			@Override
			public void onClick(DialogInterface dlg, int arg) {
				dlg.dismiss();
			}
		});
        return builder.create();
    }
    
    public static Dialog MessageDialog(Activity context, String text, DialogInterface.OnClickListener handler)
    {
    	Builder builder = new AlertDialog.Builder(context);
        builder.setMessage(text);
        builder.setCancelable(false);
        builder.setPositiveButton(context.getResources().getString(R.string.Ok), handler);
        return builder.create();
    }

    public static Dialog QuestionDialog(Activity context, String text, DialogInterface.OnClickListener handler_yes, DialogInterface.OnClickListener handler_no)
    {
    	Builder builder = new AlertDialog.Builder(context);
        builder.setMessage(text);
        builder.setCancelable(false);
        builder.setNegativeButton(context.getResources().getString(R.string.No), handler_no);
        builder.setPositiveButton(context.getResources().getString(R.string.Yes), handler_yes);
        return builder.create();
    }
}

File: EIDSS v5/android/EIDSS/EidssAndroidHelpers.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace EIDSS
{
    public class EidssAndroidHelpers
    {
        public static Dialog ErrorDialog(Activity context, string text)
        {
            var builder = new AlertDialog.Builder(context);
            builder.SetMessage(text);
            builder.SetCancelable(false);
            builder.SetPositiveButton(context.Resources.GetString(Resource.String.Ok), (sender, args) =>
            {
                var dialog = (AlertDialog)sender;
                dialog.Dismiss();
            });
            return builder.Create();
        }

        public static Dialog MessageDialog(Activity context, string text, EventHandler<DialogClickEventArgs> handler)
        {
            var builder = new AlertDialog.Builder(context);
            builder.SetMessage(text);
            builder.SetCancelable(false);
            builder.SetPositiveButton(context.Resources.GetString(Resource.String.Ok), handler);
            return builder.Create();
        }

        public static Dialog QuestionDialog(Activity context, string text, EventHandler<DialogClickEventArgs> handler_yes, EventHandler<DialogClickEventArgs> handler_no)
        {
            var builder = new AlertDialog.Builder(context);
            builder.SetMessage(text);
            builder.SetCancelable(false);
            builder.SetNegativeButton(context.Resources.GetString(Resource.String.No), handler_no);
            builder.SetPositiveButton(context.Resources.GetString(Resource.String.Yes), handler_yes);
            return builder.Create();
        }
    }
}